[PPM-286] Remove AGENT_PING_REQUEST/RESPONSE

VladyslavTupikin opened the pull request:
Need to remove next messages and all related source code:

```
ACTION_CONTROL_AGENT_PING_REQUEST
ACTION_CONTROL_AGENT_PING_RESPONSE
```

The main reason is: these vendor-specific messages no longer be useful and should be removed.

https://jira.prplfoundation.org/browse/PPM-286

Commit:
common: tlvf: Remove AGENT_PING_ message

Need to remove next messages and all
related code because those vendor-specific
message no needed anymore.

ACTION_CONTROL_AGENT_PING_REQUEST
ACTION_CONTROL_AGENT_PING_RESPONSE

Also remove all deprecated code
from autogenerated files.

https://jira.prplfoundation.org/browse/PPM-286

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
common: bcl: Remove deprecated error code

Removed deprecated error code,
fixed numeration for the rest of
error codes.

https://jira.prplfoundation.org/browse/PPM-286

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
agent: son_slave: Remove process_keep_alive method

Remove process_keep_alive method and all
related variables.

https://jira.prplfoundation.org/browse/PPM-286

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
agent: son_slave: Remove case from VS message handler

Need to remove case for handling
ACTION_CONTROL_AGENT_PING_RESPONSE message.

https://jira.prplfoundation.org/browse/PPM-286

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
controller: son_master: Remove case for VS message

Need to renmove case for ACTION_CONTROL_AGENT_PING_REQUEST
message from handler.

https://jira.prplfoundation.org/browse/PPM-286

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

morantr commented:
Please do all the PING message removal commits on a single PR (#1534).

VladyslavTupikin commented:
@morantr  I have 2 separate issues, created 2 separate pull requests.

mariomaz commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -251,53 +251,6 @@ bool slave_thread::work()
-
-            send_cmdu_to_controller(cmdu_tx);
-            keep_alive_retries++;
-            master_last_seen = now;
```
I think `keep_alive_retries`and `master_last_seen` member variables are not used any more and should be removed too

VladyslavTupikin replied:
fixed

mariomaz commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -251,53 +251,6 @@ bool slave_thread::work()
 
-void slave_thread::process_keep_alive()
-{
-    if (!config.enable_keep_alive || !son_config.slave_keep_alive_retries ||
```
I think `enable_keep_alive` and `slave_keep_alive_retries` configuration attributes are not used any more and should be removed too

VladyslavTupikin replied:
fixed

mariomaz reviewed (approved):
I think some variables and configuration parameters are not used any more after removing the PING messages and thus they should be removed too.

Commit:
common: tlvf: Remove slave_keep_alive_retries

Need to remove depricated variable
slave_keep_alive_retries.

Also remove deprecated code from
autogenerated files.

https://jira.prplfoundation.org/browse/PPM-286

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
controller: database: Remove deprecated variable

Need to remove deprecated variable
from database ehader file slave_keep_alive_retries.

https://jira.prplfoundation.org/browse/PPM-286

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
agent: son_slave: Remove deprecated variable

Need to remove usage of deprecated
variable slave_keep_alive_retries.

Also remove the corresponding variables from bcl and controller.

https://jira.prplfoundation.org/browse/PPM-286

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
cpmmon: bcl: Remove unused variable

Need to remove deprecated variable
enable_keep_alive.

Also remove that variable from agent
source code.

https://jira.prplfoundation.org/browse/PPM-286

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
agent: son_slave: Remove deprecated variable

Need to remove deprecated variable
master_last_seen.

https://jira.prplfoundation.org/browse/PPM-286

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

arnout commented on framework/platform/bpl/include/bpl/bpl_err.h:
```diff
@@ -45,7 +45,6 @@ namespace bpl {
 /* 15 */ ERROR_CODE(BH_STOPPED)                                                             \
 /* 16 */ ERROR_CODE(SLAVE_CONNECTING_TO_BACKHAUL_MANAGER)                                   \
-/* 17 */ ERROR_CODE(SLAVE_MASTER_KEEP_ALIVE_TIMEOUT)                                        \
 /* 18 */ ERROR_CODE(SLAVE_INVALID_MASTER_SOCKET)                                            \
```
Note that this changes the numbering of everything that comes after....

VladyslavTupikin replied:
Fixed commented numeration for rest of error codes.

arnout commented on common/beerocks/bcl/include/bcl/beerocks_config_file.h:
```diff
@@ -74,7 +74,6 @@ class config_file {
         std::string monitor_ap_active_threshold_B;
         std::string monitor_ap_idle_stable_time_sec;
         std::string monitor_disable_initiative_arp;
-        std::string slave_keep_alive_retries;
```
(nitpick) the commit that remove this variable could be squashed together, with a commit log like this:

```
agent: remove deprecated slave_keep_alive_retries

Also remove the corresponding variables from bcl and controller.
```

VladyslavTupikin replied:
Fixed
